### PR TITLE
Refactor colors to reflect styleguide

### PR DESCRIPTION
**Pull request by ywsang**
## Overview

Updated colors and names in `UIColor+Shared` to match the style guide, update color/font use, and update shadows throughout to match final screen designs on Figma.

## Changes Made

Biggest changes in: 
- `UIColor+Shared.swift` and replacing usages throughout app 
- `Histogram.swift`
- ClassList calendar header
- `FilterViewController.swift`
- a lot of little font changes and styling to match designs better

## Test Coverage

Verified visually that app matches designs. 

## Related PRs or Issues 

Addressed: #181, #143 

## Screenshots

**New Histogram**

**Updated Classes Page**

<summary> Empty state fix </summary>

| iPhone 8 | iPhone 6s |
:-------------------------:|:-------------------------:
|  |  |

### CODE REVIEW

File: Uplift/Views/Home/GymListItemCell.swift [hunk 1]
```diff
@@ -26,18 +26,16 @@ class GymListItemCell: ListItemCollectionViewCell<Gym> {
...
+        contentView.layer.backgroundColor = UIColor.primaryWhite.cgColor
+        contentView.layer.borderColor = UIColor.gray01.cgColor
+        contentView.layer.borderWidth = 1.0
 
         // SHADOWING
-        contentView.layer.shadowColor = UIColor.fitnessBlack.cgColor
-        contentView.layer.shadowOffset = CGSize(width: 0.0, height: 15.0)
-        contentView.layer.shadowRadius = 5.0
-        contentView.layer.shadowOpacity = 0.1
+        contentView.layer.shadowColor = UIColor(red: 0.15, green: 0.15, blue: 0.37, alpha: 0.1).cgColor
```
Comment (kevinchan159) [inline Uplift/Views/Home/GymListItemCell.swift]: is this shadowColor only used once?

Comment (ywsang) [reply Uplift/Views/Home/GymListItemCell.swift]: yes sir

File: Uplift/Controllers/ClassListViewController.swift [hunk 2]
```diff
@@ -217,15 +217,16 @@ extension ClassListViewController: UICollectionViewDelegate, UICollectionViewDat
             var dayOfWeekLabelTextColor: UIColor?
 
             if dateForCell < currDate {
-                dateLabelTextColor = .fitnessMediumGrey
-                dayOfWeekLabelTextColor = .fitnessMediumGrey
+                dateLabelTextColor = .gray02
```
Comment (chalo2000) [inline Uplift/Controllers/ClassListViewController.swift]: could there be a more descriptive name for this color?

File: Uplift/Controllers/FilterViewController.swift [hunk 3]
```diff
@@ -97,20 +97,20 @@ class FilterViewController: UIViewController, RangeSeekSliderDelegate {
 
         // START TIME SLIDER SECTION
         fitnessCenterStartTimeDivider = UIView()
-        fitnessCenterStartTimeDivider.backgroundColor = .fitnessLightGrey
+        fitnessCenterStartTimeDivider.backgroundColor = .gray01
```
Comment (chalo2000) [inline Uplift/Controllers/FilterViewController.swift]: ditto

File: Uplift/Controllers/FilterViewController.swift [hunk 4]
```diff
@@ -97,20 +97,20 @@ class FilterViewController: UIViewController, RangeSeekSliderDelegate {
...
         contentView.addSubview(fitnessCenterStartTimeDivider)
 
         startTimeTitleLabel = UILabel()
         startTimeTitleLabel.sizeToFit()
-        startTimeTitleLabel.font = ._12LatoBlack
-        startTimeTitleLabel.textColor = .fitnessDarkGrey
+        startTimeTitleLabel.font = ._12MontserratBold
+        startTimeTitleLabel.textColor = .gray04
```
Comment (chalo2000) [inline Uplift/Controllers/FilterViewController.swift]: i think you get the point now

**Comment by ywsang** [comment]
> Just a small concern on the names gray01-gray05

Used these color names specifically to match the color names that design used in the style guide to make it a lot easier to figure out what color to make things

**Pull request merged**